crates/fuel-core/src/state/in_memory/memory_store.rs

```diff
@@ -62,6 +69,23 @@ impl<Description> MemoryStore<Description>
 where
     Description: DatabaseDescription,
 {
+    fn create_view(&self) -> MemoryView<Description> {
+        // Lock all tables at the same time to have consistent view.
+        let locks = self
+            .inner
+            .iter()
+            .map(|lock| lock.lock().expect("Poisoned lock"))
```

Comment:
Can we explain in the `expect` why we assume this isn't poisoned? I'm curious, but it's also best practice.

Reply:
The code below works in the same way=D We could replace it with another lock that doesn't require expect in a separate RP.

Reply:
That would be even better than what I suggested.